springwebclient-benchmark/src/main/java/benchmark/springwebclient/Engine.java

```diff
@@ -0,0 +1,91 @@
+
+import com.ss.benchmark.httpclient.common.HttpClientEngine;
+import io.netty.channel.ChannelOption;
+import io.netty.handler.timeout.ReadTimeoutHandler;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.client.reactive.ReactorClientHttpConnector;
+import org.springframework.web.reactive.function.BodyInserters;
+import org.springframework.web.reactive.function.client.ClientResponse;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.core.publisher.Mono;
+import reactor.netty.http.client.HttpClient;
+import reactor.netty.resources.ConnectionProvider;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+public class Engine implements HttpClientEngine {
+
+    private WebClient webClient;
+
+    @Override
+    public void createClient(String host, int port) {
+        String baseUrl = url(host, port);
+        webClient = WebClient
+                .builder()
+                .clientConnector(new ReactorClientHttpConnector(createReactorNettyClient()))
+                .baseUrl(baseUrl)
+                .build();
+    }
+
+    @Override
+    public String blockingGET(String path) {
+        return webClient
+                .get()
+                .uri(path)
+                .exchange()
+                .flatMap(res -> consumeResponse(res))
+                .doOnError(t -> System.err.println("Failed requesting server: " + t.getMessage()))
+                .block();
+    }
+
+    @Override
+    public String blockingPOST(String path, String body) {
+        return webClient
+                .method(HttpMethod.POST)
+                .uri(path)
+                .body(BodyInserters.fromPublisher(Mono.just(body), String.class))
+                .exchange()
+                .flatMap(res -> consumeResponse(res))
+                .doOnError(t -> System.err.println("Failed requesting server: " + t.getMessage()))
+                .block();
+    }
+
+    @Override
+    public CompletableFuture<String> nonblockingGET(String path) {
+        return webClient
+                .get()
+                .uri(path)
+                .exchange()
+                .flatMap(res -> consumeResponse(res))
+                .toFuture();
+    }
+
+    @Override
+    public CompletableFuture<String> nonblockingPOST(String path, String body) {
+        return webClient
+                .method(HttpMethod.POST)
+                .uri(path)
+                .body(BodyInserters.fromPublisher(Mono.just(body), String.class ))
+                .exchange()
+                .flatMap(res -> consumeResponse(res))
+                .toFuture();
+    }
+
+    private Mono<String> consumeResponse(ClientResponse res) {
+        if(res.rawStatusCode() != 200) {
+            Mono.error(new RuntimeException("Unexpected response code : " + res.rawStatusCode()));
+        }
+        return res.bodyToMono(String.class);
+    }
+
+    private HttpClient createReactorNettyClient() {
```

Comment:
How about making this a static method in `reactornetty-benchmark` and re-using it here.  We're doing [the same thing](https://github.com/sharathsrinivasa/httpclientbenchmark/blob/52dfee1a9ad442794ad101f3a2cfeec7c8ec0859/reactornetty-benchmark/src/main/java/benchmark/reactornetty/Engine.java#L23) there.

Reply:
Initially, did not wanted to cross pollinate the utility code between different modules. But, this makes sense. Will do.

Reply:
Will skip it. This would make springwebclient module unnecessarily depend on reactornetty module. IMO, code duplicity is OK.

Reply:
That is a good point.